### PR TITLE
Documentation for runtime-plugin

### PR DESCRIPTION
**Pull request by jimil749**
This PR adds basic developer manual/documentation for runtime plugins. Let me know if I missed something or if there's any scope for improvement! :sweat_smile: 

Signed-off-by: Jimil Desai <jimildesai42@gmail.com>

### CODE REVIEW

File: docs/content/en/docs/concepts/plugins.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    "github.com/hashicorp/go-plugin"
+)
+
+// Assume this implements the user.Manager interface
+type Manager struct{}
+
+// Handshake hashicorp go-plugin handshake
+var Handshake = plugin.HandshakeConfig{
+	ProtocolVersion:  1,
+	MagicCookieKey:   "BASIC_PLUGIN",
```
Comment (labkode) [inline docs/content/en/docs/concepts/plugins.md:58]: @jimil749 would make sense that the protocol version and Magic* contents are constants defined in the Reva plugin package that can be used?

Comment (jimil749) [reply docs/content/en/docs/concepts/plugins.md:58]: Yes, that would be in-fact better alternative. The developer can then just import the package and use the exported consts. :+1:

Comment (labkode) [reply docs/content/en/docs/concepts/plugins.md:58]: Awesome

Comment (jimil749) [reply docs/content/en/docs/concepts/plugins.md:58]: I'll update the documentation once we get #1970 merged. (There might be more changes)

**Review by labkode — APPROVED** [review]

**Pull request merged**